cmd/gateway/azure/gateway-azure.go

```diff
@@ -553,7 +658,9 @@ func (a *azureObjects) ListObjects(ctx context.Context, bucket, prefix, marker,
 
 	result.Objects = objects
 	result.Prefixes = prefixes
-	result.NextMarker = marker
+	if marker != "" {
+		result.NextMarker = azureMarkerPrefix + marker
+	}
 	result.IsTruncated = (marker != "")
```

Comment:
As you check for `marker != nil`, you could assign `result.IsTruncated = true` in the `if` block.